ride-cleansing/src/main/java/org/apache/flink/training/exercises/ridecleansing/RideCleansingExercise.java

```diff
@@ -70,17 +72,23 @@ public JobExecutionResult execute() throws Exception {
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
 
         // set up the pipeline
-        env.addSource(source).filter(new NYCFilter()).addSink(sink);
+        env.addSource(source)
+                .filter(taxiRide -> GeoUtils.isInNYC(taxiRide.startLon, taxiRide.startLat)
```

Comment:
You already wrote the logic for filtering in NYCFilter below, try to use an instance of that class in this .filter() call.

Reply:
In the ppt slide 28 there is a requirement to rewrite the filter as .filter as a lambda function